### PR TITLE
Remove unnessesary call to text from std.conv.convError

### PR DESCRIPTION
**Pull request by JackStouffer**
Removes an extra array allocation.

### CODE REVIEW

File: std/conv.d [hunk 1]
```diff
@@ -73,26 +73,44 @@ class ConvException : Exception
...
-    return source.empty ? "end of input" : text("'", source.front, "'");
-}
-
 private auto convError(S, T)(S source, string fn = __FILE__, size_t ln = __LINE__)
 {
-    return new ConvException(
-        text("Unexpected ", convError_unexpected(source),
-             " when converting from type "~S.stringof~" to type "~T.stringof),
-        fn, ln);
+    if (source.empty)
```
Comment (wilzbach) [inline std/conv.d]: How about?

```
enforce!ConvException(!source.empty, ...)
```

**Comment by JackStouffer** [comment]
@klickverbot Changed

**Comment by John-Colvin** [comment]
Style guide says binary operators (`~` in this case) should have space around.

Other than that, LGTM

**Comment by JackStouffer** [comment]
Ping @klickverbot @wilzbach

**Comment by JackStouffer** [comment]
Ping @klickverbot @wilzbach

**Comment by JackStouffer** [comment]
Ping @wilzbach @quickfur.

**Comment by JackStouffer** [comment]
Thanks

**Pull request merged**